erna/scripts/upload.py

```diff
@@ -74,17 +74,27 @@ def xml(config, fact_tools_version, name, comment, xml_file):
 
 @click.command()
 @click.option('--config', help='Yaml file containing database credentials')
+@click.option('--version', help='Set the version, if not the jar is executed to obtain it')
 @click.argument('jar-file')
-def jar(config, jar_file):
+def jar(config, version, jar_file):
 
     config = load_config(config)
 
     log.debug('Connecting to database')
     database.init(**config['processing_database'])
 
-    output = sp.check_output(['java', '-jar', jar_file]).decode()
-    version = re.match('.*Version ([0-9]+\.[0-9]+\.[0-9]+)', output).groups()[0]
-    log.info('Found FACT Tools version "{}"'.format(version))
+    output = sp.check_output(['java', '-jar', jar_file, '--version']).decode()
+
+    if version is None:
+        version_info = {}
+        for line in output.splitlines():
+            k, value = line.split(':')
+            version_info[k.strip()] = value.strip()
+
+        version = version_info['git description']
```

Comment:
Maybe test beforehand whether the key exists and create a nice error while failing if it doesn't exists? For example ask the user to add their own version and restart the program.